ec/resources/dnsroute/tsig.go

```diff
@@ -24,169 +25,243 @@ func ResourceTsig() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"account_number": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "Account Number for the customer if not already specified in the provider configuration."},
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Account Number associated with the customer whose 
+				resources you wish to manage. This account number may be found 
+				in the upper right-hand corner of the MCC.`},
 			"alias": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: "Alias."},
+				Description: "Indicates a brief description for the TSIG key."},
 			"key_name": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "tsig key name"},
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Identifies the key on the master name server and 
+				our Route name servers. This name must be unique.`},
 			"key_value": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "tsig value"},
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Identifies a hash value through which our name 
+				servers will be authenticated to a master name server.`},
 			"algorithm_name": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "tsig encryption type:[HMAC-MD5,HMAC-SHA1,HMAC-SHA256,HMAC-SHA384,HMAC-SHA224,HMAC-SHA512]"},
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Identifies a cryptographic hash function name. 
+				Options: HMAC-MD5 | HMAC-SHA1 | HMAC-SHA256 | HMAC-SHA384 | 
+				HMAC-SHA224 | HMAC-SHA512`},
 		},
 	}
 }
 
-func ResourceTsigCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	var diags diag.Diagnostics
+func ResourceTsigCreate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Initialize Route DNS Service
 	accountNumber := d.Get("account_number").(string)
+	config := m.(**api.ClientConfig)
+	routeDNSService, err := buildRouteDNSService(**config)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	// Construct TSIG Object
```

Comment:
Could use an "ExpandTSIG" function and also use for Update?

Reply:
Done! Thank you!